sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala

```diff
@@ -290,8 +284,8 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
         df,
         dataSource.createSink(outputMode),
         outputMode,
-        useTempCheckpointLocation = useTempCheckpointLocation,
-        recoverFromCheckpointLocation = recoverFromCheckpointLocation,
+        useTempCheckpointLocation = true,
```

Comment:
AFAIK, It is not suitable to use temporary checkpoint location for other sinks beside "console", temporary directory will be deleted after process is finished. So the ability of checkpoint recovery is lost.

Also for `ConsoleSink`, there's no consistency and failure recovery guarantee, so it should not set  `recoverFromCheckpointLocation` to `true`.

Correct me if I'm wrong.